test/NetTopologySuite.Samples.Console/Technique/SplitPolygonAtDateline.cs

```diff
@@ -0,0 +1,802 @@
+        /// approach may be to shift all coordinates to the 0° to +360° range. This will allow polygons with larger widths.  This will get implemented
+        /// in a future update if required. However, it really depends on how the user inputs the coordinates. If the input already has coordinates in
+        /// the 0° to +360° range then that is straightforward to detect when a geometry crosses the dateline. But if the inputs are all in the standard
+        /// -180° to 0° to +180° range then the assumption has to be made if the geometry closes towards the prime meridian or towards the dateline.
+        /// The current algorithm determines this by simply calculting which critical meridian is the closest.
+        /// https://github.com/NetTopologySuite/NetTopologySuite/discussions/496
+        /// https://github.com/NetTopologySuite/NetTopologySuite/blob/develop/test/NetTopologySuite.Samples.Console/Operation/Polygonize/SplitPolygonExample.cs
+        /// https://github.com/NetTopologySuite/NetTopologySuite/blob/84aef3c6a6cd3c2bfa5e3034ba161555cef5a117/test/NetTopologySuite.Samples.Console/Operation/Polygonize/SplitPolygonExample.cs#L24
+        /// https://github.com/nettopologysuite/nettopologysuite/issues/129
+        /// https://github.com/DotSpatial/DotSpatial/blob/e30421d6ee5b0cdf2b1fa01f95f8aabc50c1997a/Source/DotSpatial.Data/Shape.cs#L611
+        /// https://github.com/DotSpatial/DotSpatial/issues/1029
+        /// https://github.com/NetTopologySuite/ProjNet4GeoAPI/wiki/Well-Known-Text
+        /// https://desktop.arcgis.com/en/arcmap/10.3/manage-data/editing-fundamentals/creating-and-editing-multipart-polygons.htm
+        /// https://desktop.arcgis.com/en/arcmap/10.3/guide-books/map-projections/what-happens-to-features-at-180-dateline-.htm
+        /// </remarks>
+        public static Geometry ToPolyExOp(List<Coordinate> inpCoords, string inpProj4Wkt, OgcGeometryType outType, double outTrimGap, double outDensifyResolution, bool isAttFixOutInvPolygons)
+        {
+            Debug.Print("$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$");
+
+            // ###################################################################################################################
+            // input coordinate checks
+            // ###################################################################################################################
+            if (inpCoords == null)
+            {
+                throw new Exception("Null input coordinates.");
+            }
+
+            // check the minimum number of input coordinates for an output geometry
+            switch (outType)
+            {
+                case OgcGeometryType.LineString:
+                    if (inpCoords.Count < 2)
+                        throw new Exception("Input list has fewer than 2 coordinates. To create a polyline, the minimum is 2 coordinates.");
```

Comment:
ArgumentOutOfRange (of maybe InvalidOperationException?) here

Reply:
I think I agree with your assessment here and will change my code. However, on my work related code I like to tell the user exactly what is wrong. At times I've encountered the ArgumentOutOfRange exception and wondered "ok what is the range then?".

Reply:
you can leave the message. I see your point and ArgumentException can be used too. 
I must admit that, once you don't use the base `Exception` class, the proper exception to use can be arguable , so feel free to use the exception class you want. Often I use `InvalidOperationException`, but not sure I can be taken as an example :)

Reply:
Sure thing. I am learning on the go.

Reply:
I think I can use something like:

`throw new ArgumentOutOfRangeException(nameof(count), count, "Must be non-negative.");`